File: 1894-merge-strings-alternately/1894-merge-strings-alternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        n1=len(word1)
        n2=len(word2)
        word = ""
        count1=0
        count2=0

        while count1!= n1 and count2!= n2:
            count = count1 + count2
            word += word1[count1]
            count1+=1
            word += word2[count2]
            count2+=1
        word = word +''.join(list(word1[count1:]))
        word = word +''.join(list(word2[count2:]))
        return word

File: 1894-merge-strings-alternately/1894-merge-strings-alternately.c


char * mergeAlternately(char * word1, char * word2)
{
    int n1 = strlen(word1);
    int n2 = strlen(word2);
    int maxlen = n1 + n2;

    char* result = (char*)malloc((maxlen + 1) * sizeof(char));
    
    int i=0 , j=0, k=0 ; 

    while(i < n1 && j<n2 ){
        result[k++] = word1[i++];
        result[k++] = word2[j++];
    }
    while(i<n1){
        result[k++] = word1[i++];
    }
    while(j<n2){
        result[k++] = word2[j++];
    }
    
    result[k] = '\0'; // Null terminate the string
    return result;
}